engine/src/main/java/io/camunda/zeebe/engine/state/migration/DbMigratorImpl.java

```diff
@@ -76,22 +77,30 @@ public DbMigratorImpl(
   @Override
   public void runMigrations() {
     logPreview(migrationTasks);
-
     final var executedMigrations = new ArrayList<MigrationTask>();
+    zeebeDbContext.runInTransaction(
```

Comment:
❌ Not sure especially related to https://github.com/camunda/zeebe/issues/14976 and discussion here https://camunda.slack.com/archives/C06D6MWDCMD/p1705444740113089

whether it is good to use only one transaction. In general, it makes sense if we want to ensure that everything has been migrated when we are done "atomically"

Reply:
I could be wrong of course

Reply:
Yep, learning from https://github.com/camunda/zeebe/pull/16090, we should definitely _not_ run just a single transaction :+1:

Reply:
Done in the latest rebase.